recipes/arrow-adbc/meta.yaml

```diff
@@ -0,0 +1,203 @@
+{% set version = "0.1.0" %}
+
+package:
+  name: arrow-adbc-split
+  version: {{ version }}
+
+source:
+  url: https://dist.apache.org/repos/dist/release/arrow/apache-arrow-adbc-{{ version }}/apache-arrow-adbc-{{ version }}.tar.gz
+  sha256: a50f3dac985883042152938bdd69c843708fd50508e0f15ba3a179619c11f7b9
+  patches:
+    # Backport https://github.com/apache/arrow-adbc/issues/304, drop in 0.2.0
+    - patches/0001-fix-c-driver-postgresql-define-ntohll-etc-for-macOS.patch
+
+build:
+  number: 0
+
+outputs:
+  - name: adbc-driver-manager-cpp
+    script: build-cpp.sh                                                         # [not win]
+    script: build-cpp.bat                                                        # [win]
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+      host:
+      run:
+    test:
+      commands:
+        - test -f $PREFIX/include/adbc.h                                         # [unix]
+        - test -f $PREFIX/include/adbc_driver_manager.h                          # [unix]
+        - test -d $PREFIX/lib/cmake/AdbcDriverManager/                           # [unix]
+        - test -f $PREFIX/lib/pkgconfig/adbc-driver-manager.pc                   # [unix]
+        - test ! -f $PREFIX/lib/libadbc_driver_manager.a                         # [unix]
+        - test -f $PREFIX/lib/libadbc_driver_manager.so                          # [linux]
+        - test -f $PREFIX/lib/libadbc_driver_manager.dylib                       # [osx]
```

Comment:
I suggest checking for a specific major SO name in order to guard against ABI breakages.